### PR TITLE
Replace KUBERNETES_PROVIDER with --provider

### PR DESCRIPTION
**Pull request by oomichi**
Since 07c6da22fe9f97ab44a02e67a7c574b49998605d, KUBERNETES_PROVIDER
has been deprecated. So this PR replaces KUBERNETES_PROVIDER with
the option --provider to specify a provider on e2e test.

### CODE REVIEW

**Comment by cblecker** [comment]
I'm guessing this was from back when @fejta converted hack/e2e.go into a shim for kubetest.
/unassign @calebamiles @cblecker
/assign @fejta @crimsonfaith91

File: contributors/devel/e2e-tests.md [hunk 1]
```diff
@@ -76,8 +76,7 @@ of ways to run e2e tests to a canonical way: `hack/e2e.go`.
 You can run an end-to-end test which will bring up a master and nodes, perform
 some tests, and then tear everything down. Make sure you have followed the
 getting started steps for your chosen cloud platform (which might involve
-changing the `KUBERNETES_PROVIDER` environment variable to something other than
-"gce").
+changing the --provider option value to something other than "gce").
```
Comment (crimsonfaith91) [inline contributors/devel/e2e-tests.md]: `--provider flag value`?

File: contributors/devel/e2e-tests.md [hunk 2]
```diff
@@ -432,7 +431,7 @@ at a custom host directly:
 export KUBECONFIG=/path/to/kubeconfig
 export KUBE_MASTER_IP="127.0.0.1:<PORT>"
 export KUBE_MASTER=local
-go run hack/e2e.go -- -v --test
+go run hack/e2e.go -- -v --test --provider=local
```
Comment (crimsonfaith91) [inline contributors/devel/e2e-tests.md]: What is the default value for `--provider`? If it is `local`, we may not need the flag.

File: contributors/devel/e2e-tests.md [hunk 3]
```diff
@@ -115,7 +114,7 @@ GINKGO_PARALLEL=y go run hack/e2e.go -- -v --test --test_args="--ginkgo.skip=\[S
 # You can also specify an alternative provider, such as 'aws'
 #
 # e.g.:
-KUBERNETES_PROVIDER=aws go run hack/e2e.go -- -v --build --up --test --down
+go run hack/e2e.go -- -v --build --up --test --down --provider=aws
```
Comment (crimsonfaith91) [inline contributors/devel/e2e-tests.md]: Is there a reason putting the flag behind? Will this work: `go run hack/e2e.go -- --provider=aws -v --build --up --test --down`?

File: contributors/devel/e2e-tests.md (hunk 1 above)
Comment (oomichi) [inline contributors/devel/e2e-tests.md]: @crimsonfaith91 Thanks for your review. That is better, I will apply it soon.

File: contributors/devel/e2e-tests.md (hunk 3 above)
Comment (oomichi) [inline contributors/devel/e2e-tests.md]: Yeah, that works fine. I confirmed it like:

$ go run hack/e2e.go -- --provider=skeleton -v --test  --test_args="--ginkgo.focus=blank"
2017/08/17 09:35:03 e2e.go:76: Calling kubetest --provider=skeleton -v --test --test_args=--ginkgo.focus=blank...
2017/08/17 09:35:03 util.go:131: Running: ./cluster/kubectl.sh --match-server-version=false version
2017/08/17 09:35:03 util.go:133: Step './cluster/kubectl.sh --match-server-version=false version' finished in 76.603564ms
2017/08/17 09:35:03 util.go:131: Running: ./hack/e2e-internal/e2e-status.sh
[..]
Ran 1 of 675 Specs in 11.386 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 674 Skipped PASS

Ginkgo ran 1 suite in 11.583277352s
Test Suite Passed
2017/08/17 09:35:15 util.go:133: Step './hack/ginkgo-e2e.sh --ginkgo.focus=blank' finished in 11.813172925s
2017/08/17 09:35:15 e2e.go:80: Done
$

I will move the option as you pointed.

File: contributors/devel/e2e-tests.md (hunk 2 above)
Comment (oomichi) [inline contributors/devel/e2e-tests.md]: The option doesn't have the default value like:

test/e2e/framework/test_context.go
  219         flag.StringVar(&TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, vagrant, etc.)")

So we need the flag here.

Comment (crimsonfaith91) [inline contributors/devel/e2e-tests.md]: Sure! Thanks for getting back.

**Pull request merged**